shim/src/epoll/epoll.c

```diff
@@ -67,9 +76,44 @@ int epoll_get_ready(int epfd, demi_qtoken_t *qts, struct demi_event **evs)
         ev = epoll_get_next(epfd, ev);
     }
 
+    // Check if any of the sends have received acks
+    struct demi_event *send_ev = epoll_get_head(SEND_EPFD);
+    while (send_ev != NULL)
+    {
+        if ((send_ev->used) && (send_ev->qt != (demi_qtoken_t)-1))
+        {
+            qts[nevents] = send_ev->qt;
+            evs[nevents] = send_ev;
+            nevents++;
+        }
+
+        send_ev = epoll_get_next(SEND_EPFD, send_ev);
+    }
+
```

Comment:
Could you use a for loop to limit the scope of `send_ev` to the for loop itself?

Reply:
Sounds good! I also refactored epoll_get_ready into a more general function, so there is no code replication for the procedure that get the ready events from the regular epoll table and the send events.